### PR TITLE
Improve buffer size handling / avoid MAX_PATH

### PR DESCRIPTION
**Pull request by ThFabba**
MAX_PATH is a Win32 concept that makes no sense in the kernel (unless specifically interacting with a Win32 process). Plus, fixed size buffers should be avoided anyway where possible.

### CODE REVIEW

File: ntoskrnl/config/cmhvlist.c [hunk 1]
```diff
@@ -160,6 +159,16 @@ CmpAddToHiveFileList(IN PCMHIVE Hive)
         return Status;
     }
 
+    Length = sizeof(OBJECT_NAME_INFORMATION) + 256 * sizeof(WCHAR);
```
Comment (ThFabba) [inline ntoskrnl/config/cmhvlist.c]: Meh, should probably ask ZwQueryObject for the length and avoid the fixed size while I'm at it.

File: ntoskrnl/io/pnpmgr/pnpmgr.c [hunk 2]
```diff
@@ -2885,11 +2880,23 @@ IopActionConfigureChildServices(PDEVICE_NODE DeviceNode,
...
+      RegKey.Buffer = ExAllocatePoolWithTag(PagedPool,
+                                            RegKey.MaximumLength,
+                                            TAG_IO);
+      if (RegKey.Buffer == NULL)
+      {
+          IopDeviceNodeSetFlag(DeviceNode, DNF_DISABLED);
+          return STATUS_INSUFFICIENT_RESOURCES;
+      }
+
+      RtlAppendUnicodeToString(&RegKey, ENUM_ROOT);
```
Comment (HBelusca) [inline ntoskrnl/io/pnpmgr/pnpmgr.c]: Potential bug: the `#define ENUM_ROOT` does **NOT** currently specify the trailing backslash: https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=ENUM_ROOT

File: ntoskrnl/config/cmhvlist.c (hunk 1 above)
Comment (HBelusca) [reply ntoskrnl/config/cmhvlist.c]: 256 --> MAXIMUM_FILENAME_LENGTH

File: ntoskrnl/config/cmlazy.c [hunk 3]
```diff
@@ -297,6 +296,16 @@ CmpCmdHiveOpen(IN POBJECT_ATTRIBUTES FileAttributes,
             return STATUS_OBJECT_PATH_SYNTAX_BAD;
         }
 
+        Length = sizeof(OBJECT_NAME_INFORMATION) + 256 * sizeof(WCHAR);
```
Comment (HBelusca) [inline ntoskrnl/config/cmlazy.c]: 256 --> MAXIMUM_FILENAME_LENGTH

File: ntoskrnl/io/pnpmgr/pnpmgr.c [hunk 4]
```diff
@@ -1853,7 +1853,7 @@ IopGetParentIdPrefix(PDEVICE_NODE DeviceNode,
     }
 
     /* 1. Try to retrieve ParentIdPrefix from registry */
-    KeyNameBufferLength = FIELD_OFFSET(KEY_VALUE_PARTIAL_INFORMATION, Data[0]) + MAX_PATH * sizeof(WCHAR);
+    KeyNameBufferLength = FIELD_OFFSET(KEY_VALUE_PARTIAL_INFORMATION, Data[0]) + 32 * sizeof(WCHAR);
```
Comment (HBelusca) [inline ntoskrnl/io/pnpmgr/pnpmgr.c]: What does this "32" value corresponds to? some sizeof() ? some length of a string?

File: ntoskrnl/config/cmsysini.c [hunk 5]
```diff
@@ -1357,7 +1357,7 @@ VOID
 NTAPI
 CmpInitializeHiveList(VOID)
 {
-    WCHAR FileBuffer[MAX_PATH], RegBuffer[MAX_PATH];
+    WCHAR FileBuffer[64], RegBuffer[64];
```
Comment (HBelusca) [inline ntoskrnl/config/cmsysini.c]: What are these arbitrary-looking "64" values?

File: ntoskrnl/ex/handle.c [hunk 6]
```diff
@@ -1292,7 +1292,7 @@ BOOLEAN ExpKdbgExtHandle(ULONG Argc, PCHAR Argv[])
     EXHANDLE ExHandle;
     PLIST_ENTRY Entry;
     PEPROCESS Process;
-    WCHAR KeyPath[MAX_PATH];
+    WCHAR KeyPath[256];
```
Comment (HBelusca) [inline ntoskrnl/ex/handle.c]: 256 --> MAXIMUM_FILENAME_LENGTH

Comment (ThFabba) [reply ntoskrnl/ex/handle.c]: This is not a file name.

File: ntoskrnl/config/cmsysini.c (hunk 5 above)
Comment (ThFabba) [reply ntoskrnl/config/cmsysini.c]: 64 is exactly as arbitrary as MAX_PATH, and both are large enough to hold these values.

File: ntoskrnl/io/pnpmgr/pnpmgr.c (hunk 4 above)
Comment (ThFabba) [reply ntoskrnl/io/pnpmgr/pnpmgr.c]: Will change to `sizeof(L"12345678&12345678")`, which is what this function writes in there... nobody else should write these anyway.

File: ntoskrnl/config/cmsysini.c (hunk 5 above)
Comment (ThFabba) [reply ntoskrnl/config/cmsysini.c]: The longest strings that should need to fit in here:
`RegBuffer`: "\Registry\Machine\Hardware" = 26
`FileBuffer`: "\SystemRoot\System32\Config\Hardware"  = 36
However using these strings to determine the size would make `CmpMachineHiveList` kind of pointless.

**Pull request merged**